### ISSUE TITLE
TOC alignment issue with macros (which contain links) and typeset

### ISSUE THREAD

**Issue by Vasperous** [squidfunk/mkdocs-material#5590]
### Context

I have been really looking forward to using the `typeset` plugin with my documentation as many of my headers are referring to code or objects.  Showing them in the TOC as `<code>` makes it much easier to read.

### Bug description

When enabling macros and typeset the TOC seems to not correctly align, specifically when the header is set to be bold and the macros replacement text is a URL.

While putting together this description I started deleting items like extensions etc. in my `mkdocs.yml` file.  I saw various changes in the rendering of what I outlined below.  My 'issue' went away when I removed the `toc_depth` option from the `toc` extension.  I am not sure if that is a bug with the TOC extension or with the way macros/URLs are being handled.  Could use some direction on that and since I am not sure I left my bug report as is.

In my `.md` file I have the following:

```
### {{rq_host}}

The host object that is not bold.

### **{{rq_host}}**

The host object that is bold...

### **{{rq_test}}**

This is a test of the macros with a non-link/url feature
```

If I only include the `### {{rq_host}}` which is a URL but not bold then everything is displayed without issue:

![image](https://github.com/squidfunk/mkdocs-material/assets/8703632/fb18d3d4-c9ba-4d0f-ac54-03b4dcff6b13)

If we only include `### **{{rq_test}}**` which is just plain text and bold then everything is displayed without issue:

![image](https://github.com/squidfunk/mkdocs-material/assets/8703632/89350605-fa2f-46fb-880d-46f9c5066310)

Only when we use the combination of a link and the bold do we see an issue with the TOC alignment `### **{{rq_host}}**`:

![image](https://github.com/squidfunk/mkdocs-material/assets/8703632/e0cba13c-38b5-4cac-8be2-6e8a15a09fb4)

**Note:** it looks like the whole page is also bold (in previous examples it was not).

For reference, contents of the external yaml file used by macros:
```
rq_host: "[`host`](#test)"
rq_test: "Test information"
```

I tried reducing down the number of features in my `mkdocs.yml` and after deleting all of the content and navigation features I got the below which is what I am seeing in my other documentation (it was at this moment that I started playing around with the extensions and added the second paragraph at the top):

![image](https://github.com/squidfunk/mkdocs-material/assets/8703632/2d7aa9d5-0dea-477a-8b36-02d2b117d4db)

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Typeset](https://squidfunk.github.io/mkdocs-material/reference/#built-in-typeset-plugin)
- 

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11664572/example.zip)

The example.zip didn't include the external macros folder so I unzipped it, added it and then re-zipped the files.

### Steps to reproduce

1. Create site
2. Use macros, typeset and toc features/plugins
3. Create macros page which references a URL
4. include the macros link in the documentation and make the text bold
5. Build documentation

### Browser

Other

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting, and for the excellent reproduction! Confirmed, it's definitely related to the typeset plugin.

**Comment by squidfunk** [comment]
Fixed in [`2a1023da4`](https://github.com/squidfunk/mkdocs-material-insiders/commit/2a1023da4c55f4115ffb77507ed9ea6cb90266e0). Thanks again for the excellent reproduction. There were two problems:

1. The regular expression to find the permalink at the end of the headline was too greedy. If an author provided a custom link as part of the headline, the opening tag of that link was matched with the closing tag of the permalink. Thus all content from the opening link was lost, leaving only the `<strong>` opening tag, which explains the bold text.

2. Author-provided anchor links were not removed at all. This is now fixed by inverting order and not limiting anchor link removal to the link at the start. This problem was also fixed in the commit.

**Comment by squidfunk** [comment]
Released as part of [9.1.16+insiders-4.36.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.36.0).
